### DISCUSSION TITLE
Slider RFC

### DISCUSSION THREAD

**Discussion by serifluous** [twilio-labs/paste#3314] [Review request]
# Request for comment (RFC): Slider

_tl;dr Please leave feedback on:_

- _The [Slider spec](https://docs.google.com/document/d/1IYdOLIueQkmq_lZomlFm71Ae3FaST3_kX0XCRIowQqI/edit?usp=sharing) in the Google doc_
- _The API (using [react-aria's useSlider](https://react-spectrum.adobe.com/react-aria/useSlider.html)) in this discussion_
- _The [Slider design](https://www.figma.com/file/1AqHqB9kTvNISG2vwDQEst/Slider-design-explorations?type=design&node-id=148-19449&mode=design&t=5NRxMnKMw8FZnsHt-4) in the Figma file._ 

## Spec

Paste is adding a Slider component and would love input from all roles.

Check out the proposed spec and guidelines here: https://docs.google.com/document/d/1IYdOLIueQkmq_lZomlFm71Ae3FaST3_kX0XCRIowQqI/edit?usp=sharing

A few highlights:

- We are recommending Slider be used primarily for choosing imprecise values. When a precise value is needed, use number Input (currently in development #3312), optionally pairing it with a Meter (currently on the roadmap).
- Slider follows the guidelines of all our other form elements (e.g., Input, Select) and requires a label, uses help text, and discourages the disabled state.
- Related components include number Input, [Radio Button Group](https://paste.twilio.design/components/radio-button-group), [Radio Group](https://paste.twilio.design/components/radio-group), and [Select](https://paste.twilio.design/components/select). **If you have any questions about which component is right for your use case, please let us know in this discussion!**

## API

We are proposing that Slider wraps [react-aria's useSlider](https://react-spectrum.adobe.com/react-aria/useSlider.html).

**Please let us know how well (or not) its features work for your product.**

## Design

**Review the design proposal below by dropping comments in the Figma file:** https://www.figma.com/file/1AqHqB9kTvNISG2vwDQEst/Slider-design-explorations?type=design&node-id=148-19449&mode=design&t=5NRxMnKMw8FZnsHt-4

Does the proposed design work for existing use cases in your product? Is there important information or interaction missing in the design that's needed for your use case?

![image](https://github.com/twilio-labs/paste/assets/3821840/35f41feb-0531-4aea-9575-b53b7053075c)

![image](https://github.com/twilio-labs/paste/assets/3821840/27855c51-857e-4695-a725-7f964e0292af)

Thanks!

**Comment by TheSisb** [comment]
The component has been released and the window for comment has been closed.
